### PR TITLE
add release channels and support levels

### PR DESCRIPTION
**Pull request by astoycos**
Add stable and experimental release channels and Core, Extended, Implementation-specific support levels in line with the gateway-api.

Currently all of the apis we provide are experimental since they are v1alpha1 version.

Also add documentation on the new channels + Support levels and update our NPEP process slightly.

### CODE REVIEW

File: site-src/enhancements.md [hunk 1]
```diff
@@ -9,40 +11,61 @@ process for the main Kubernetes project:
...
+  implementation details have not been agreed to yet.
+* **Implementable:** The goals and implementation details described by this GEP
+  have consensus but have not been fully implemented yet.
+* **Experimental:** This GEP has been implemented and is part of the
+  "Experimental" release channel. Breaking changes are still possible, up to
+  and including complete removal and moving to `Rejected`.
+* **Standard:** This GEP has been implemented and is part of the
+  "Standard" release channel. It should be quite stable.
+* **Deferred:** We do not currently have bandwidth to handle this GEP, it
+  may be revisited in the future.
```
Comment (npinaeva) [inline site-src/enhancements.md:29]: replace GEP with NPEP :)

File: site-src/enhancements.md [hunk 2]
```diff
@@ -53,21 +76,69 @@ It is likely that throughout this process, you will discuss a variety of
 alternatives. Be sure to document all of these in the NPEP, and why we decided
 against them. At this stage, the NPEP should be targeting the "Implementable" stage.
 
-### 6. Standard: Make API changes
-With the NPEP marked as "Implementable", it is time to actually make the proposed changes in our API. 
+### 6. Experimental: Make API changes as "Experimental
```
Comment (npinaeva) [inline site-src/enhancements.md:79]: add double quote in the end

File: site-src/enhancements.md [hunk 3]
```diff
@@ -53,21 +76,69 @@ It is likely that throughout this process, you will discuss a variety of
...
 only, but in most cases, some API changes will also be required.
 
-When updating NPEP from "Implementable" to "Standard", status update may be the 
+It is important that every new feature of the API is marked as "Experimental" when it is introduced. Within the API, we
+use <network-policy-api:experimental> tags to denote experimental fields. Within Golang packages (conformance tests,
+CLIs, e.t.c.) we use the experimental Golang build tag to denote experimental functionality.
+
+Some other requirements must be met before marking a NPEP Experimental:
+
+the graduation criteria to reach Standard MUST be filled out
```
Comment (npinaeva) [inline site-src/enhancements.md:91]: should this be a list?

File: site-src/enhancements.md [hunk 4]
```diff
@@ -53,21 +76,69 @@ It is likely that throughout this process, you will discuss a variety of
...
+a proposed probationary period (see next section) must be included in the NPEP and approved by maintainers.
+
+When updating NPEP from "Implementable" to "Standard", status update may be the
```
Comment (npinaeva) [inline site-src/enhancements.md:94]: from "Implementable" to "Experimental"?

File: site-src/enhancements.md [hunk 5]
```diff
@@ -53,21 +76,69 @@ It is likely that throughout this process, you will discuss a variety of
...
 only required change, but also feel free to link any API, docs, etc. changes that
 were made as an implementation for a given NPEP.
 
 Before changes are released they MUST be documented. NPEPs that have not been
 both implemented and documented before a release cut off will be excluded from
 the release.
 
+#### Probationary Period
```
Comment (npinaeva) [inline site-src/enhancements.md:102]: maybe this should be a subsection of `6. Experimental: Make API changes as "Experimental` ?

**Review by npinaeva — COMMENTED** [review]
nice addition to what we have!

File: site-src/enhancements.md (hunk 5 above)
Comment (astoycos) [reply site-src/enhancements.md:102]: It is :) notice the heading size 👍

**Comment by astoycos** [comment]
/hold 

(to prevent accidental merge)

File: site-src/enhancements.md [hunk 6]
```diff
@@ -9,40 +11,61 @@ process for the main Kubernetes project:
...
    historical reference.
 
+## NPEP Status
+
+Every NPEP has a `Status` field, that is updated with the progress of NPEP.
+Current statuses are:
+
+* **Provisional:** The goals described by this NPEP have consensus but
+  implementation details have not been agreed to yet.
+* **Implementable:** The goals and implementation details described by this GEP
```
Comment (npinaeva) [inline site-src/enhancements.md:21]: GEP is still here :)

**Review by Dyanngg — COMMENTED** [review]
LGTM overall

**Review by Dyanngg — APPROVED** [review]

**Review by tssurya — APPROVED** [review]
API changes lgtm. I have not tested them (will do soon)
This is great that we are having two sets of CRDs, standard and experimental.
thank you Andrew!

**Pull request merged**